ui/imports/shared/popups/send/SendModal.qml

```diff
@@ -211,11 +166,35 @@ StatusDialog {
         if(popup.preSelectedSendType !== Constants.SendType.Unknown) {
             store.setSendType(popup.preSelectedSendType)
         }
-        if ((popup.preSelectedHoldingType > Constants.TokenType.Native) &&
-                (popup.preSelectedHoldingType < Constants.TokenType.Unknown)) {
-            tokenListRect.browsingHoldingType = popup.preSelectedHoldingType
-            if (!!popup.preSelectedHoldingID) {
-                d.setSelectedHoldingId(popup.preSelectedHoldingID, popup.preSelectedHoldingType)
+        if (!!popup.preSelectedHoldingID
+                && popup.preSelectedHoldingType > Constants.TokenType.Native
+                && popup.preSelectedHoldingType < Constants.TokenType.Unknown) {
+
+            if (popup.preSelectedHoldingType === Constants.TokenType.ERC20) {
+                const entry = ModelUtils.getByKey(
+                                assetsAdaptor.outputAssetsModel, "tokensKey",
+                                popup.preSelectedHoldingID)
+                d.selectedHoldingType = Constants.TokenType.ERC20
```

Comment:
That's good question. I didn't go too deep into this aspect in this PR. If you think it should be solved in this PR, I will take a look.

Reply:
Ah, ok.. if you think we can do it in a sperate one, ok! However, since here you are touching the "holdings" selection topic, maybe we can just do the cleanup of this part all together, but, you can chose!

Reply:
Fixing this snowballs into a bunch of other changes. I have it in mind and will address in upcoming PRs, step by step.